src/components/textarea/textarea.ts

```diff
@@ -0,0 +1,359 @@
+    start: number,
+    end: number,
+    selectMode: 'select' | 'start' | 'end' | 'preserve' = 'preserve'
+  ) {
+    this.input.setRangeText(replacement, start, end, selectMode);
+    this.value = this.input.value;
+  }
+
+  /* blazorSuppress */
+  public override scrollTo(options?: ScrollToOptions | undefined): void;
+  public override scrollTo(x: number, y: number): void;
+  public override scrollTo(x?: unknown, y?: unknown): void {
+    x !== undefined && y !== undefined
+      ? this.input.scrollTo(x as number, y as number)
+      : this.input.scrollTo(x as ScrollToOptions);
+  }
+
+  protected resolvePartNames(base: string) {
+    return {
+      [base]: true,
+      prefixed: this.prefixes.length > 0,
+      suffixed: this.suffixes.length > 0,
+      filled: !!this.value,
+    };
+  }
+
+  @watch('value', { waitUntilFirstUpdate: true })
+  protected async valueChanged() {
+    await this.updateComplete;
+    this.setAreaHeight();
+  }
+
+  @watch('rows', { waitUntilFirstUpdate: true })
+  @watch('resize', { waitUntilFirstUpdate: true })
+  protected setAreaHeight() {
+    if (this.resize === 'auto') {
+      this.input.style.height = 'auto';
+      this.input.style.height = `${this.input.scrollHeight}px`;
+    } else {
+      Object.assign(this.input.style, { height: undefined });
+    }
+  }
+
+  protected handleInput() {
+    this.value = this.input.value;
+    this.emitEvent('igcInput', { detail: this.value });
+  }
+
+  protected handleChange() {
+    this.value = this.input.value;
+    this.emitEvent('igcChange', { detail: this.value });
+  }
+
+  protected valueSlotChange() {
+    const value = extractText(this.projected);
+
+    if (value.length) {
+      this.value = value.join('\r\n');
+    }
+  }
+
+  protected renderValueSlot() {
+    return html`<slot
+      style="display: none"
+      @slotchange=${this.valueSlotChange}
+    ></slot>`;
+  }
+
+  protected renderPrefix() {
+    return html`<div part="prefix">
+      <slot name="prefix"></slot>
+    </div>`;
+  }
+
+  protected renderSuffix() {
+    return html`<div part="suffix">
+      <slot name="suffix"></slot>
+    </div>`;
+  }
+
+  protected renderLabel() {
+    return this.label
+      ? html`<label part="label" for=${this.id || this.inputId}
+          >${this.label}</label
+        >`
+      : nothing;
+  }
+
+  protected renderStandard() {
+    return html`
+      ${this.renderLabel()}
+      <div part=${partNameMap(this.resolvePartNames('container'))}>
+        ${this.renderPrefix()} ${this.renderInput()} ${this.renderSuffix()}
+      </div>
+      <div part="helper-text" .hidden=${this.helperText.length < 1}>
+        <slot name="helper-text"></slot>
+      </div>
+    `;
+  }
+
+  protected renderMaterial() {
+    return html`
+      <div
+        part=${partNameMap({
+          ...this.resolvePartNames('container'),
+          labelled: this.label,
+        })}
+      >
+        <div part="start">${this.renderPrefix()}</div>
+        ${this.renderInput()}
```

Comment:
Can we move the notch before the input? Otherwise, the notch is displayed over the textarea therefore the text can't be scrolled.